### PR TITLE
Adding platform controller models in Editor play mode

### PR DESCRIPTION
**Pull request by keveleigh**
This PR adds the ability to use the platform's controller models in the Editor. Previously, this scenario was limited to displaying the controller overrides.

The DLL source can be found here: https://github.com/Microsoft/MixedRealityToolkit/pull/122

### CODE REVIEW

File: Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs [hunk 1]
```diff
@@ -44,6 +50,11 @@ public class MotionControllerVisualizer : MonoBehaviour
         private Dictionary<uint, MotionControllerInfo> controllerDictionary = new Dictionary<uint, MotionControllerInfo>(0);
         private List<uint> loadingControllers = new List<uint>();
 
+#if UNITY_EDITOR_WIN
+        [DllImport("MotionControllerModel")]
+        private static extern int TryGetMotionControllerModel([In] uint controllerId, [Out] out ulong outputSize, [Out] out IntPtr outputBuffer);
```
Comment (fieldsJacksonG) [inline Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: bool

File: Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs [hunk 2]
```diff
@@ -262,18 +261,32 @@ private IEnumerator LoadControllerModel(InteractionSource source)
 
                     reader.ReadBytes(fileBytes);
                 }
+#else
+                IntPtr controllerModel = new IntPtr();
+                ulong outputSize = 0;
+
+                if (TryGetMotionControllerModel(source.id, out outputSize, out controllerModel) > 0)
```
Comment (fieldsJacksonG) [inline Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: "> 0" not needed if returning bool

File: Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs [hunk 3]
```diff
@@ -262,18 +261,32 @@ private IEnumerator LoadControllerModel(InteractionSource source)
 
                     reader.ReadBytes(fileBytes);
                 }
+#else
+                IntPtr controllerModel = new IntPtr();
+                ulong outputSize = 0;
```
Comment (fieldsJacksonG) [inline Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: ulong would match up to LONGLONG in C++, but your plugin is UINT32.  Potential overflow

Comment (fieldsJacksonG) [reply Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/types

**Comment by cre8ivepark** [comment]
Tested with Unity 2017.2.0f3-MRTP. Confirmed both default controller model and prefab override works well! Awesome work @keveleigh !
2017-10-27 12_19_47-mixed reality portal

File: Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs (hunk 3 above)
Comment (keveleigh) [reply Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: Oof, looks like I made some changes while testing that didn't get reverted. Thanks for catching!

**Pull request merged**